### PR TITLE
ThreadCounter

### PR DESCRIPTION
**Pull request by Stanislava88**
Here i have solved the first three tasks for thread counters.

### CODE REVIEW

**Comment by freekman** [comment]
The first task can be separated in a new package to be easily found

File: src/com/clouway/counter/ThreadCounter.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Override
+    public void run() {
+        System.out.println(Thread.currentThread().getName() + "starting!");
+        try {
+            for (int i = 0; i < countTo && !Thread.currentThread().isInterrupted(); i++) {
+                count = i;
+                System.out.println(count);
+
+                Thread.sleep(1000);
```
Comment (freekman) [inline src/com/clouway/counter/ThreadCounter.java]: What is the difference between ```Thread.sleep(1000)``` and ```Thread.sleep(1000L)```

File: src/com/clouway/counter/ThreadCounter.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            }
+        } catch (InterruptedException e) {
+            System.out.println("Thread interrupted");
```
Comment (freekman) [inline src/com/clouway/counter/ThreadCounter.java]: it is good practice to ```return``` if an ```InterruptedException``` is caught

File: src/com/clouway/counter/ThreadCounter.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        }
+        System.out.println(count);
```
Comment (freekman) [inline src/com/clouway/counter/ThreadCounter.java]: ``` System.out.println(count);``` is this for the 
- ```След прочитането на символа да се спира нишката и да се извежда до коя стойност е стигнало отброяването.```

File: src/com/clouway/counter/counters/Demo.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package com.clouway.counter.counters;
```
Comment (freekman) [inline src/com/clouway/counter/counters/Demo.java]: i really dont like how the packages are  ```/counter/counters/......../thecounter```

File: src/com/clouway/counter/counters/ThreadCounter.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package com.clouway.counter.counters;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class ThreadCounter extends Thread {
+    private int from;
```
Comment (freekman) [inline src/com/clouway/counter/counters/ThreadCounter.java]: private final

File: src/com/clouway/counter/ThreadCounter.java (hunk 1 above)
Comment (Stanislava88) [inline src/com/clouway/counter/ThreadCounter.java]: if i use without L it's convert during runtime to a long

File: src/com/clouway/counter/ThreadCounter.java (hunk 3 above)
Comment (Stanislava88) [inline src/com/clouway/counter/ThreadCounter.java]: Yes

File: src/com/clouway/counter/counters/ThreadCounter.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    private ThreadCounter thread;
+
+    public ThreadCounter(int countFrom, int countTo) {
+        this.from = countFrom;
+        this.to = countTo;
+    }
+
+    @Override
+    public void run() {
+        for (int i = from; i <= to && !isInterrupted(); i++) {
```
Comment (freekman) [inline src/com/clouway/counter/counters/ThreadCounter.java]: for (int i = from; i <= to ```&& !isInterrupted()```; i++) 
- this is the first time is see someone to use && or || in a for()
please remove it

File: src/com/clouway/counter/ThreadCounter.java (hunk 1 above)
Comment (freekman) [inline src/com/clouway/counter/ThreadCounter.java]: you use it with ```L``` in some places and without ```L``` in other it is better if you make it with ```L``` every where

**Comment by Stanislava88** [comment]
done

File: src/com/clouway/counters/ThreadCounter.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    private ThreadCounter thread;
+
+    public ThreadCounter(int countFrom, int countTo) {
+        this.from = countFrom;
+        this.to = countTo;
+    }
+
+    @Override
+    public void run() {
+        for (int i = from; i <= to && !isInterrupted(); i++) {
```
Comment (freekman) [inline src/com/clouway/counters/ThreadCounter.java]: .

File: src/com/clouway/pausingexecution/ThreadCounter.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    public ThreadCounter(int from, int to) {
+        this.from = from;
+        this.to = to;
+    }
+
+    @Override
+    public void run() {
+        try {
+            for (int i = from; i <= to && !isInterrupted(); i++) {
```
Comment (freekman) [inline src/com/clouway/pausingexecution/ThreadCounter.java]: .

**Comment by freekman** [comment]
LGTM

**Pull request merged**